### ISSUE TITLE
Attack range bug, mod interactions

### ISSUE THREAD

**Issue by Uthael** [FTBTeam/FTB-Interactions#823]
## Modpack Version: 
I heard it's 1.9.1, Where do I see that? 

## Bug Report + Is It Repeatable? Steps to Reproduce:
**TL;DR - Attack range is bugged when modified.**
Details:
Playing on the Dirtcraft multiplayer server. Made a support ticket to the server network. After some time, they figured: "Its plugin / mod interaction issue.. I don't think we can fix this.". So, I decided to copy the most important parts from the conversation over to here...:

--------------------------------------------------------------------------------------
Ring of Arcana (#6227) in the SWRG mode almost constantly pushes mobs outside of their attack range. This is fine, but I wanteded to be able to reach them. I've tried Cyclic's Block Reach enchant (chest piece), Astral Sorcery's perk named Combat Focus and Botania's Ring of Far Reach. 

I've tested the situation extensively, both in singleplayer and multiplayer, with different results. 

Singleplayer: 
In each box, there's a zombie who aggroes on the player and is trying to pathfind through the corner. I was hugging the protruded strip, standing on the same height as the zombies. Marked the attack range with signs. 2 signs mean the range is in-between. 
1 - the default vanilla range. 
2 - the range with the perk (not enough for reaching when the mob is pushed away) 
3 - single Ring of Far Reach 
4 - Ring and the perk 
5, 6 and 7 are with equipped chest piece. It overwrites the rest. 
![rangetest](https://user-images.githubusercontent.com/2061046/68330930-5eb1d180-00d4-11ea-88d3-7779e2e1ed92.png)

Multiplayer server: 
- I can't test without the perk as it's a major hassle to make Irradiant Stars there. I spent quite a few in singleplayer. - the perk, however works. All the small % perk bonuses that increase it, work too. 
- Ring of Far Reach doesn't work. 
- the enchant doesn't work. Unequipping the enchant also resets the attack range to vanilla until relog or until the enchant is reequipped. 

As for the block placement, the ring works and the enchant works. They don't stack, enchant owerwrites the ring. Same bug with unequipping the enchant.

--------------------------------------------------------------------------------------
Tested it some more. 
It's not just ring-of-reach issue, it affects any increases in attack range. Is it some hack protection or something? Maybe this will help in troubleshooting the issue: The One Probe displays mob's buffs and its HP as a number if you sneak while the mob is within range. Also, the vanilla mechanic of checking if something is within your range is to look at the attack indicator right under your cursor. If the mob is within your swing range you see the indicator.

tests: 
- Used Astral Sorcery to increase my range as much as I could.
![Range_increase_in_Astral_Sorcery](https://user-images.githubusercontent.com/2061046/68333541-fe715e80-00d8-11ea-9b57-ee8842718223.jpg)

- Edge of melee range as registered by the server
![Range_registered_by_the_server](https://user-images.githubusercontent.com/2061046/68333547-0204e580-00d9-11ea-8307-07c766a3d1d9.jpg)

- Range registered on the client (Those perks and one Ring of Far Reach)
![Range_registered_on_the_client](https://user-images.githubusercontent.com/2061046/68333552-03cea900-00d9-11ea-9e5e-177729cadc37.jpg)

## Mod/s Affected: 
Cyclic, Astral Sorcery, Botania, maybe more

**Comment by Saereth** [comment]
Oof yeah this is one of those mod interaction edge cases where they are all just trying to do similar things and order of implementation/recalculations get really tricky, can't really offer much help here as getting 3-4 different mods to all agree on how to do this is probably just not gonna happen. My only advice is just stick with the most reliable sources and try not to stack from other mods.
